judge/views/submission.py

```diff
@@ -47,9 +53,29 @@ class SubmissionDetailBase(LoginRequiredMixin, TitleMixin, SubmissionMixin, Deta
     def get_object(self, queryset=None):
         submission = super(SubmissionDetailBase, self).get_object(queryset)
         if not submission.can_see_detail(self.request.user):
-            raise PermissionDenied()
+            raise SubmissionPermissionDenied(submission)
         return submission
 
+    def get(self, request, *args, **kwargs):
+        try:
+            return super(SubmissionDetailBase, self).get(request, *args, **kwargs)
```

Comment:
```suggestion
            return super().get(request, *args, **kwargs)
```
This is Python 2 syntax. We're slowly migrating to Python 3 syntax, so any new code should just use `super()`.

Reply:
Alright, pushed.